pkgs/tools/package-management/appimagekit/nix.patch

```diff
@@ -172,3 +172,16 @@ index cf5fd5c..4f48dbc 100644
  #include "shared.h"
  
  #if HAVE_LIBARCHIVE3 == 1 // CentOS
+diff --git a/src/appimagetool.c b/src/appimagetool.c
+index 69beaa1..b797ae6 100644
+--- a/src/appimagetool.c
++++ b/src/appimagetool.c
+@@ -348,7 +345,7 @@ void extract_arch_from_text(gchar *archname, const gchar* sourcename, bool* arch
+ void guess_arch_of_file(const gchar *archfile, bool* archs) {
+     char line[PATH_MAX];
+     char command[PATH_MAX];
+-    sprintf(command, "/usr/bin/file -L -N -b %s", archfile);
++    sprintf(command, "file -L -N -b %s", archfile);
```

Comment:
This relies on `file` being present in $PATH. How about doing it with `substituteInPlace`. Here is an example:
https://github.com/NixOS/nixpkgs/blob/60a09a9dcc41bc92478a52fa6c05a2074b943f8c/pkgs/tools/X11/xdg-utils/default.nix#L47-L48

Reply:
Good point. The code does require the `file` command to be in the PATH elsewhere, but I solved this by wrapping `appimagetool`.